bin/src/utils.dart

```diff
@@ -0,0 +1,238 @@
+import 'dart:io';
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/file_system/file_system.dart' as fs;
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:analyzer/source/package_map_resolver.dart';
+import 'package:analyzer/source/pub_package_map_provider.dart';
+import 'package:analyzer/source/sdk_ext.dart';
+import 'package:analyzer/src/dart/sdk/sdk.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/java_io.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer/src/generated/source_io.dart';
+import 'package:path/path.dart' as path;
+
+final RegExp newLinePartOfRegexp = new RegExp('\npart of ');
+final RegExp partOfRegexp = new RegExp('part of ');
+
+Iterable<ClassElement> moduleClassesWithoutNames(
+    AnalysisContext context, Directory sdkDir) {
+  final entryPoints = getPackageEntryPoints();
+  final sources = parseSources(context, entryPoints);
+  verifyAnalysis(context, sources);
+
+  return parseLibraries(context, sources)
+      .expand(getSubclassesOfLifecycleModule)
+      .where((element) => element.getField('name') == null);
+}
+
+Map<Source, List<ClassElement>> groupClassesBySource(
+    List<ClassElement> classes) {
+  final Map<Source, List<ClassElement>> results = {};
+
+  classes.forEach((c) {
+    if (results.containsKey(c.source)) {
+      results[c.source].add(c);
+    } else {
+      results[c.source] = [c];
+    }
+  });
+
+  return results;
+}
+
+void writeGettersForSource(Source source, List<ClassElement> classes) {
+  // Sort descending to modify the bottom classes before the top
+  classes.sort((a, b) => b.computeNode().end - a.computeNode().end);
```

Comment:
This is pretty naive but I find it works well avoiding "concurrent modification" of things like files.

Reply:
I think it's a great solution!! 🌟